homeassistant/components/zwave_me/__init__.py

```diff
@@ -113,7 +113,7 @@ def __init__(self, controller, device):
     def device_info(self) -> DeviceInfo:
         """Return device specific attributes."""
         return DeviceInfo(
-            identifiers={(DOMAIN, self._attr_unique_id)},
+            identifiers={(DOMAIN, self.device.deviceIdentifier)},
```

Comment:
I don't believe there's a good way to migrate the existing device entries, as the user may have customized them differently. We wouldn't know which one to apply to the new device entry.

We should remove the old device entries though. Please add that and tests for that.

https://github.com/home-assistant/core/blob/b093c2840b97c1585ec26bf2431d3d708ad6dc4f/homeassistant/helpers/device_registry.py#L269-L275

https://github.com/home-assistant/core/blob/b093c2840b97c1585ec26bf2431d3d708ad6dc4f/homeassistant/helpers/device_registry.py#L520-L522